dScripts/02_server/Map/AG/NpcCowboyServer.cpp

```diff
@@ -23,4 +24,17 @@ void NpcCowboyServer::OnMissionDialogueOK(Entity* self, Entity* target, int miss
 	} else if (missionState == eMissionState::READY_TO_COMPLETE || missionState == eMissionState::COMPLETE_READY_TO_COMPLETE) {
 		inventoryComponent->RemoveItem(14378, 1);
 	}
+
+	// Next up hide or show the samples based on the mission state
+	int32_t visible = 1;
+	if (missionState == eMissionState::READY_TO_COMPLETE || missionState == eMissionState::COMPLETE_READY_TO_COMPLETE) {
+		visible = 0;
+	}
+
+	auto spawners = Game::zoneManager->GetSpawnersByName("PlungerGunTargets");
+	for (auto* spawner : spawners) {
```

Comment:
Nitpick: Does auto* spawner need to be mutable here?

Reply:
ehh i get it but its overkill tbqh

Reply:
tbf I am very much in the 'const all the things!' school of thought. Shows you whether your access is read/only or not and makes debugging easier. But I won't block if you really don't want to do it.

Reply:
I get the mindset and will keep it in mind